ImmunityReborn/Assets/_TeamAmnesia/Scripts/ShieldCollisions.cs

```diff
@@ -13,49 +13,111 @@ public class ShieldCollisions : MonoBehaviour
     [field: SerializeField]
     public float ShieldKnockback { get; private set; }
 
-    private void OnTriggerEnter( Collider other )
+    [SerializeField]
+    private float PerfectParryWindowDuration;
+
+    private void Start()
     {
-        // Notes for future implementation: 
-        // Debug shows that this trigger gets called once on click (hits the player I guess)
-        // Then 3 more times as 3 bullets touch the shield (to see which projectile should get deleted)
+        playerStateMachine = GetComponentInParent<PlayerStateMachine>();
+    }
 
-        if ( other.gameObject.tag == typeTag ) // Ranged projectile block code here
-        {
-            Destroy( other.gameObject );
-            playerStateMachine.MemoryGauge.EarnMemoryGauge( 10 );
-        }
 
-        if ( typeTag == "MeleeType" && other.gameObject.TryGetComponent( out WeaponDamager weaponDamager ) && 
-             weaponDamager.DamageType == DamageType.Melee )
+    private void OnTriggerEnter( Collider other )
+    {
+        // If we actually get a Trigger from something that can do damage, not an enemy model moving into Shield
+        if ( other.gameObject.TryGetComponent( out WeaponDamager weaponDamager ) ) 
         {
+            // Get the attacker data for counterattack calculations
             GameObject attacker = weaponDamager.CharacterCollider.gameObject;
 
-            if (attacker.TryGetComponent(out Health health))
+            // Melee Damage + Melee Shields = Reward Player For Melee Mechanic Handling
+            if ( weaponDamager.DamageType == DamageType.Melee && shieldTypeCollided == "MeleeType" )
             {
-                health.DealDamage(0); // Hits the shield so it does 0 dmg 
+
+                if ( attacker.TryGetComponent( out Health health ) )
+                {
+                    health.DealDamage( 0 ); // Hits the shield so it does 0 dmg 
```

Comment:
@chrisyandev do you know why we need this ? The codeblock here is basically when they succesfully block the attack. Not sure why we need damage code where it does 0 damage. I left it for now

Reply:
For the Alpha, it was an easy way to stagger the enemy. You can remove it since we have melee counterattack now.